src/Hazelcast.Net/Clustering/ClusterMembers.cs

```diff
@@ -315,7 +315,7 @@ public async Task RemoveConnectionAsync(MemberConnection connection)
                         // otherwise, we're really disconnecting: flip _connected, and change the state
                         _connected = false;
                         _logger.IfDebug()?.LogDebug("Removed connection {ConnectionId} to member {MemberId}, disconnecting.", connection.Id.ToShortString(), connection.MemberId.ToShortString());
-                        // FIXME ClusterChanged CANNOT go back to DISCONNECTED it WILL go to CONNECTED no matter what
+                        // ClusterChanged hold Connected state. It's just a transation.
```

Comment:
This FIXME exists because: the call to `ChangeState` switches the state to `Disconnected` and the state before that is expected to be either `Connected` or `ClusterChanged` BUT as far as I understand, `ClusterChanged` can only lead to `Connected` (if we connected successfully to the new cluster) or shutdown - I don't think there is a way to go from `ClusterChanged` to `Disconnected` - and therefore `ClusterChanged` should not be in the list of expected types here - so... either we understand and fix it, or it should remain a warning.

Reply:
Removed `ClusterChanged` from expected states and kept the note as a warning at https://github.com/hazelcast/hazelcast-csharp-client/pull/682/commits/146cc487e6723800e0609781507b11b70e1c1c41

Reply:
Ok the FIXME comment I added because I did not understand why `ClientState.ClusterChanged` appeared in the list of valid previous states, since `ClusterChanged` is not supposed to ever transition to `Disconnected`. The correct fix IMO is to remove the `ClusterChanged` value from the list *and* to remove the comment entirely - but I'd like to understand why `ClusterChanged` was there in the first place. Do you remember,

Reply:
No, I don't unfortunately. Guessing that was a leftover from the big failover refactoring?? I kept the comment as a warning but I'll remove it anyway since we have reached natural flow of the code. There is no exception to note it down.

Reply:
I think you removed one of them but there's another FIXME exactly for the same thing somewhere else in the file, any chance you can clear it too?

As a side note... my original idea was that the method that change the state would throw if the current state is not one of the expected one, that's the reason for passing the "expected states" - it's not activated yet but I think we should activate it eventually. what do you think?

Reply:
so around line 318 it's OK now but not around line 582?

Reply:
Removed the second occurrence at https://github.com/hazelcast/hazelcast-csharp-client/pull/682/commits/d316b8b7c8052dd9006ed9fbafb1ba652c50bd06. I think we should discuss throwing exception in a separated issue. WDYT? So, we can experiment whether work as expected.